### ISSUE TITLE
fromfile/tofile mess up file handle position on Python 3 (migrated from Trac #1610)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#3161]
Original ticket http://projects.scipy.org/numpy/ticket/1610
Reported 2010-09-11 by atmention:pv, assigned to unknown.

```
>>> import numpy as np
>>> f = open('foo.dat', 'w+b')
>>> f.write(np.arange(255, dtype='u1').tostring())
255
>>> f.seek(0)
0
>>> np.fromfile(f, dtype='u1', count=4)
array([0, 1, 2, 3], dtype=uint8)
>>> f.tell()
255
```

**Comment by thouis** [comment]
Comment in Trac by atmention:pv, 2010-09-11

Fixed in r8701

**Comment by thouis** [comment]
Comment in Trac by atmention:pv, 2010-09-11

... and r8707

**Comment by thouis** [comment]
Comment in Trac by atmention:rgommers, 2010-10-17

A new test is failing on Windows XP 32-bit and py25:
    ======================================================================
    ERROR: test_fromfile_tofile_seeks (test_regression.TestRegression)
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "C:\Python25\Lib\site-packages\numpy\core\tests\test_regression.py", line 1389, in test_fromfile_tofile_seeks
        ret = np.fromfile(f, count=4, dtype='u1')
    IOError: first argument must be an open file
With py31 the test passes.

**Comment by thouis** [comment]
Comment in Trac by atmention:rgommers, 2010-10-17

#1634 also looks related.

**Comment by thouis** [comment]
Comment in Trac by atmention:rgommers, 2010-10-17

This also fails with 2.6 and 2.7 on Windows.

**Comment by thouis** [comment]
Comment in Trac by atmention:pv, 2010-10-17

This is probably because of use of TemporaryFile, see http://mail.python.org/pipermail/python-bugs-list/2007-April/038012.html

Apparently, `fromfile` and `tofile` have never worked with NamedTemporaryFile objects on Windows.

#1634 is not related -- StringIO was never supported, as it does not supply FILE\* pointers.

**Comment by thouis** [comment]
Comment in Trac by atmention:pv, 2010-10-17

Suggested fix: http://github.com/numpy/numpy/compare/maintenance/1.5.x...pv:bug/1610-continued

**Comment by thouis** [comment]
Comment in Trac by atmention:rgommers, 2010-10-18

Tested on Windows py25 and py27, OS X py27. Works fine. Thanks!

**Comment by thouis** [comment]
Comment in Trac by atmention:pv, 2010-10-18

Fix for the test applied in commit:082956be and commit:98484bb6
